src/common/helper.ts

```diff
@@ -309,9 +309,14 @@ async function readProtocolStream(
   handle: string,
   path?: string
 ): Promise<Buffer> {
+  if (!isNode && path) {
+    throw new Error('Cannot write to a path outside of Node.js environment.');
+  }
+
   let eof = false;
-  let fileHandle: fs.promises.FileHandle;
+  let fileHandle: import('fs').promises.FileHandle;
```

Comment:
How is `.promises` already available synchronously? Does this not need `await`?

Reply:
No, because this is a type, so TS understands what we mean here. It's just so we can still type the `fileHandle` variable, but without an `import * as fs...` at the top of the file.

Reply:
(this line doesn't make it into the compiled output)

Reply:
Oh yeah, I totally misread it as run-time code. Got it!